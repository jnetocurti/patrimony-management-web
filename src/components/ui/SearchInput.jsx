import React, { memo } from 'react';
import { Search as SearchIcon } from 'react-feather';

import {
  Box,
  CircularProgress,
  InputAdornment,
  makeStyles,
  SvgIcon,
  TextField
} from '@material-ui/core';

import PropTypes from 'prop-types';

const useStyles = makeStyles((theme) => ({
  progress: {
    marginRight: theme.spacing(1)
  }
}));

export const SearchInputAdornment = () => {
  return (
    <InputAdornment position="start">
      <SvgIcon fontSize="small" color="action">
        <SearchIcon />
      </SvgIcon>
    </InputAdornment>
  );
};

export const SearchInputProgress = () => {
  const classes = useStyles();
  return <CircularProgress className={classes.progress} size="1.25rem" />;
};

const SearchInput = ({ callback, searchStatus, placeholder = 'Pesquisar' }) => {
  const handleChange = (e) => callback(e.target.value);

  return (
    <Box maxWidth={500}>
      <TextField
        fullWidth
        size="small"
        variant="outlined"
        onChange={handleChange}
        placeholder={placeholder}
        InputProps={{
          startAdornment: searchStatus ? (
            <SearchInputProgress />
          ) : (
            <SearchInputAdornment />
          )
        }}
      />
    </Box>
  );
};

SearchInput.propTypes = {
  callback: PropTypes.func,
  searchStatus: PropTypes.bool,
  placeholder: PropTypes.string
};

export default memo(SearchInput);
