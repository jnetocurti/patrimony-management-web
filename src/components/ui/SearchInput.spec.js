import React from 'react';

import { TextField } from '@material-ui/core';

import { shallow } from 'enzyme';
import SearchInput, {
  SearchInputAdornment,
  SearchInputProgress
} from 'src/components/ui/SearchInput';

describe('src/components/SearchInputAdornment', () => {
  it('should render correctly', () => {
    const component = shallow(<SearchInputAdornment />);
    expect(component).toMatchSnapshot();
  });
});

describe('src/components/SearchInputProgress', () => {
  it('should render correctly', () => {
    const component = shallow(<SearchInputProgress />);
    expect(component).toMatchSnapshot();
  });
});

describe('src/components/SearchInput', () => {
  let placeholder;
  let searchStatus;
  const callback = jest.fn();

  beforeEach(() => {
    searchStatus = false;
    placeholder = undefined;
  });

  it('should render with default placeholder', () => {
    const component = shallow(
      <SearchInput {...{ callback, searchStatus, placeholder }} />
    );
    expect(component).toMatchSnapshot();
  });

  it('should render with custom placeholder', () => {
    placeholder = 'Custom';
    const component = shallow(
      <SearchInput {...{ callback, searchStatus, placeholder }} />
    );
    expect(component).toMatchSnapshot();
  });

  it('should render with SearchInputProgress input adornment', () => {
    searchStatus = true;
    const component = shallow(
      <SearchInput {...{ callback, searchStatus, placeholder }} />
    );
    expect(component).toMatchSnapshot();
  });

  it('should perform the callback function when the input value changed', () => {
    const component = shallow(
      <SearchInput {...{ callback, searchStatus, placeholder }} />
    );

    component.find(TextField).simulate('change', { target: { value: 'foo' } });

    expect(callback).toHaveBeenCalledTimes(1);
    expect(callback).toHaveBeenCalledWith('foo');
  });
});
