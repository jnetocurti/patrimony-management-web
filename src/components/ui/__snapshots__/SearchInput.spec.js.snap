// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`src/components/SearchInput should render with SearchInputProgress input adornment 1`] = `
<Styled(MuiBox)
  maxWidth={500}
>
  <WithStyles(ForwardRef(TextField))
    InputProps={
      Object {
        "startAdornment": <SearchInputProgress />,
      }
    }
    fullWidth={true}
    onChange={[Function]}
    placeholder="Pesquisar"
    size="small"
    variant="outlined"
  />
</Styled(MuiBox)>
`;

exports[`src/components/SearchInput should render with custom placeholder 1`] = `
<Styled(MuiBox)
  maxWidth={500}
>
  <WithStyles(ForwardRef(TextField))
    InputProps={
      Object {
        "startAdornment": <SearchInputAdornment />,
      }
    }
    fullWidth={true}
    onChange={[Function]}
    placeholder="Custom"
    size="small"
    variant="outlined"
  />
</Styled(MuiBox)>
`;

exports[`src/components/SearchInput should render with default placeholder 1`] = `
<Styled(MuiBox)
  maxWidth={500}
>
  <WithStyles(ForwardRef(TextField))
    InputProps={
      Object {
        "startAdornment": <SearchInputAdornment />,
      }
    }
    fullWidth={true}
    onChange={[Function]}
    placeholder="Pesquisar"
    size="small"
    variant="outlined"
  />
</Styled(MuiBox)>
`;

exports[`src/components/SearchInputAdornment should render correctly 1`] = `
<WithStyles(ForwardRef(InputAdornment))
  position="start"
>
  <WithStyles(ForwardRef(SvgIcon))
    color="action"
    fontSize="small"
  >
    <Search />
  </WithStyles(ForwardRef(SvgIcon))>
</WithStyles(ForwardRef(InputAdornment))>
`;

exports[`src/components/SearchInputProgress should render correctly 1`] = `
<WithStyles(ForwardRef(CircularProgress))
  className="makeStyles-progress-1"
  size="1.25rem"
/>
`;
